### PR TITLE
Implemented: API calls to get user preferred store and ecom stores in user module

### PR DESCRIPTION
**Pull request by sanskar345**
### Related Issues

Related - https://github.com/hotwax/dxp-components/issues/193

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/oms-api/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: src/modules/user/index.ts [hunk 1]
```diff
@@ -272,7 +272,64 @@ async function setUserPreference(payload: any): Promise<any> {
...
+    if (hasError(resp)) {
+      return Promise.reject(resp.data)
+    }
+    return Promise.resolve(resp.data.userPrefValue ? resp.data.userPrefValue : '');
+  } catch (err) {
+    return Promise.reject(err)
+  }
+}
+
+async function getEComStores (facilityId: string, token: any, baseURL: string): Promise<Response> {
```
Comment (k2maan) [inline src/modules/user/index.ts:296]: Why are mentioning the returned type as `Promise<Response>`

File: src/modules/user/index.ts [hunk 2]
```diff
@@ -272,7 +272,64 @@ async function setUserPreference(payload: any): Promise<any> {
...
+      url: "performFind",
+      method: "get",
+      baseURL,
+      params,
+      headers: {
+        Authorization: 'Bearer ' + token,
+        'Content-Type': 'application/json'
+      }
+    });
+    if (hasError(resp) || resp.data.docs.length === 0) {
```
Comment (k2maan) [inline src/modules/user/index.ts:321]: This check might not be required `resp.data.docs.length === 0`

File: src/modules/user/index.ts [hunk 3]
```diff
@@ -272,7 +272,64 @@ async function setUserPreference(payload: any): Promise<any> {
...
+      baseURL,
+      headers: {
+        Authorization: 'Bearer ' + token,
+        'Content-Type': 'application/json'
+      }
+    });
+    if (hasError(resp)) {
+      return Promise.reject(resp.data)
+    }
+    return Promise.resolve(resp.data.userPrefValue ? resp.data.userPrefValue : '');
```
Comment (k2maan) [inline src/modules/user/index.ts:290]:
```suggestion
    return Promise.resolve(resp.data.userPrefValue);
```

**Review by k2maan — CHANGES_REQUESTED** [review]

**Comment by ymaheshwari1** [comment]
Closing as handled in https://github.com/hotwax/oms-api/pull/87

**Pull request closed without merging**